File: dotnet/console/Console/Scenarios/CreateXMLWithFullDocumentSignatureScenario.cs
using Lacuna.Signer.Api;
using Lacuna.Signer.Api.Documents;
using Lacuna.Signer.Api.FlowActions;
using Lacuna.Signer.Api.Users;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Console.Scenarios
{
    public class CreateXMLWithFullDocumentSignatureScenario : Scenario
    {
        /**
         * This scenario shows step by step the creation of a document
         * to the signer instance where the document is a XML file and the
         * whole document must signed.
         */
        public override async Task RunAsync()
        {
            // 1. The file's bytes must be read by the application and uploaded using the method UploadFileAsync.
            var filePath = "sample.xml";
            var fileName = Path.GetFileName(filePath);
            var file = File.ReadAllBytes(filePath);

            // 1.1 The mimeType for a xml file is "application/xml".
            var uploadModel = await signerClient.UploadFileAsync(fileName, file, "application/xml");

            // 2. Signer's server expects a FileUploadModel's list to create a document.
            var fileUploadModel = new FileUploadModel(uploadModel) { DisplayName = "XML Full Signature Sample" };

            // 3. Foreach participant on the flow, you'll need to create an instance of ParticipantUserModel.
            var participantUser = new ParticipantUserModel()
            {
                Name = "Jack Bauer",
                Email = "jack.bauer@mailinator.com",
                Identifier = "75502846369"
            };

            // 4. For a XML file it's necessary to provide for the FlowActionCreateModel a XadexOptionsModel
            //    specifying the signature type. In this case, it's only necessary to specify that the signature type
            //    it's FullXML.
            var xadesOptionsModel = new XadesOptionsModel()
            {
                SignatureType = XadesSignatureTypes.FullXml
            };

            // 5. You'll need to create a FlowActionCreateModel's instance foreach ParticipantUserModel
            //    created in the previous step. The FlowActionCreateModel is responsible for holding
            //    the personal data of the participant and the type of action that it will peform on the flow.
            //    Also, it's necessary to instantiate the propertie 'XadexOptions' with the previously created instance XadexOptionsModel.
            var flowActionCreateModel = new FlowActionCreateModel()
            {
                Type = FlowActionType.Signer,
                User = participantUser,
                XadesOptions = xadesOptionsModel
            };

            // 6. Send the document create request
            var documentRequest = new CreateDocumentRequest()
            {
                Files = new List<FileUploadModel>() { fileUploadModel },
                FlowActions = new List<FlowActionCreateModel>() { flowActionCreateModel }
            };
            var result = (await signerClient.CreateDocumentAsync(documentRequest)).First();

            System.Console.WriteLine($"Document {result.DocumentId} created");
        }
    }
}

File: nodejs/scenarios/createXmlWithFullDocumentSignatureScenario.ts
import {
    DocumentsApi,
    FileUploadModel,
    UploadApi,
    UsersParticipantUserModel,
    FlowActionsFlowActionCreateModel,
    FlowActionType,
    DocumentsCreateDocumentRequest,
    FlowActionsXadesOptionsModel,
    XadesSignatureTypes,
    getBase64,
  } from "signer-node-client";
  import { config } from "./scenario";
  
  const uploadApi = new UploadApi(config);
  const documentsApi = new DocumentsApi(config);
  const filepath = "../../samples/sample.xml";
  const filename = "sample.xml";
  
  // 1. The file's bytes must be read by the application and uploaded
  uploadApi.apiUploadsBytesPost({ bytes: getBase64(filepath) }).then((res) => {
    // 2. Define the name of the document which will be visible in the application
    const uploadModel: FileUploadModel = {
      id: res.data.id,
      name: filename,
      contentType: "application/xml",
      displayName: "XML Full Signature Sample",
    };
    // 3. For each participant on the flow, create one instance of ParticipantUserModel
    const participant: UsersParticipantUserModel = {
      name: "Jack Bauer",
      email: "jack.bauer@mailinator.com",
      identifier: "75502846369",
    };
  
    // 4. Specify the signature type
    const xadesOptionsModel: FlowActionsXadesOptionsModel = {
      signatureType: XadesSignatureTypes.FullXml
    };
  
    // 5. Create a FlowActionCreateModel instance for each action (signature or approval) in the flow.
    //    This object is responsible for defining the personal data of the participant and the type of
    //    action that he will perform on the flow.
    const flowAction: FlowActionsFlowActionCreateModel = {
      type: FlowActionType.Signer,
      user: participant,
      xadesOptions: xadesOptionsModel,
    };

    // 6. Create and send the document request
    const documentRequest: DocumentsCreateDocumentRequest = {
      files: [uploadModel],
      flowActions: [flowAction],
    };
    documentsApi.apiDocumentsPost(documentRequest).then((res) => {
      console.log("Document ", res.data[0].documentId, "created");
    });
  });
  